test/test_distributed.py

```diff
@@ -149,6 +149,14 @@ def _test_fullsync(rank, world_size, backend, q):
         except Exception as e:
             assert isinstance(e, PrefetchTimeoutError)
 
+        # Test that reset/shutdown does not hang while paused
+        dp3 = dp.fullsync()
+        it = iter(dp3)
+        next(it)
+        dp3.pause()
+        it2 = iter(dp3)  # Reset
+        next(it2)
+
```

Comment:
Is this test failing without the patch?

Reply:
Can you add a test for `DataLoader2` with fullsync?

Reply:
Yea, it fails without the patch.

Reply:
I can't really add a `DataLoader2` test because `DistributedRS` currently doesn't support `pause`....

I will have to add that separately. Let me know if I should land this as it is or add that on top of this.